it/v1/endpoints/ListPaymentsControllerISpec.scala

```diff
@@ -0,0 +1,179 @@
+          response.json shouldBe Json.toJson(expectedBody)
+          response.header("Content-Type") shouldBe Some("application/json")
+        }
+      }
+
+      val input = Seq(
+        ("AA1123A", "2018-10-01", "2019-10-01", BAD_REQUEST, NinoFormatError),
+        ("AA123456A", "2018-100-01", "2019-10-01", BAD_REQUEST, FromDateFormatError),
+        ("AA123456A", "2018-10-01", "2019-100-01", BAD_REQUEST, ToDateFormatError),
+        ("AA123456A", "2018-03-01", "2019-10-01", BAD_REQUEST, RuleFromDateNotSupportedError),
+        ("AA123456A", "2018-10-01", "2021-10-01", BAD_REQUEST, RuleDateRangeInvalidError),
+        ("AA123456A", "2018-10-01", "2018-06-01", BAD_REQUEST, RangeToDateBeforeFromDateError)
+      )
+
+      input.foreach(args => (validationErrorTest _).tupled(args))
+    }
+
+    "des service error" when {
+
+      def errorBody(code: String): String =
+        s"""{
+           |  "code": "$code",
+           |  "reason": "des message"
+           |}""".stripMargin
+
+      def serviceErrorTest(desStatus: Int, desCode: String, expectedStatus: Int, expectedBody: MtdError): Unit = {
+        s"des returns an $desCode error and status $desStatus" in new Test {
+
+          val desQueryParams: Map[String, String] = Map("dateFrom" -> from.get, "dateTo" -> to.get)
+
+          override def setupStubs(): StubMapping = {
+            AuditStub.audit()
+            AuthStub.authorised()
+            MtdIdLookupStub.ninoFound(nino)
+            DesStub.onError(DesStub.GET, desUrl, desQueryParams, desStatus, errorBody(desCode))
+          }
+
+          val response: WSResponse = await(request.get)
+          response.status shouldBe expectedStatus
+          response.json shouldBe Json.toJson(expectedBody)
+          response.header("Content-Type") shouldBe Some("application/json")
+        }
+      }
+
+      val input = Seq(
+        (BAD_REQUEST, "INVALID_IDVALUE", BAD_REQUEST, NinoFormatError),
+        (BAD_REQUEST, "INVALID_IDTYPE", INTERNAL_SERVER_ERROR, DownstreamError),
+        (BAD_REQUEST, "INVALID_REGIME_TYPE", INTERNAL_SERVER_ERROR, DownstreamError),
+        (BAD_REQUEST, "INVALID_PAYMENT_LOT", INTERNAL_SERVER_ERROR, DownstreamError),
+        (BAD_REQUEST, "INVALID_PAYMENT_LOT_ITEM", INTERNAL_SERVER_ERROR, DownstreamError),
+        (BAD_REQUEST, "INVALID_CLEARING_DOC", INTERNAL_SERVER_ERROR, DownstreamError),
+        (BAD_REQUEST, "INVALID_CORRELATIONID", INTERNAL_SERVER_ERROR, DownstreamError),
+        (BAD_REQUEST, "REQUEST_NOT_PROCESSED", INTERNAL_SERVER_ERROR, DownstreamError),
+        (BAD_REQUEST, "PARTIALLY_MIGRATED", INTERNAL_SERVER_ERROR, DownstreamError),
+        (BAD_REQUEST, "NO_DATA_FOUND", NOT_FOUND, NotFoundError),
+        (INTERNAL_SERVER_ERROR, "SERVER_ERROR", INTERNAL_SERVER_ERROR, DownstreamError),
+        (SERVICE_UNAVAILABLE, "SERVICE_UNAVAILABLE", INTERNAL_SERVER_ERROR, DownstreamError),
+        (UNPROCESSABLE_ENTITY, "INVALID_IDTYPE", INTERNAL_SERVER_ERROR, DownstreamError)
```

Comment:
Please also add tests for DES returning `INVALID_DATE_FROM` and `INVALID_DATE_TO`, since they're valid responses DES could send us (according to spec).